src/esi/Parser.cc

```diff
@@ -13,19 +13,21 @@
 #include "fatal.h"
 
 char *ESIParser::Type = NULL;
-ESIParser::Register *ESIParser::Parsers = NULL;
+std::list<ESIParser::Register *> ESIParser::Parsers;
```

Comment:
Ideally, Parsers should be a pointer (to std::list or a similar container) so that we do not implicitly rely on the container object to already be there when we need to register a parser and to still be there when we need to deregister a parser.

I would also rename Parsers to ParserMakers or similar because we do not store Parsers in this container. We store "Register"(sic) objects that make Parsers (AFAICT).

Reply:
Thanks for the reminder. Using our lookup function technique now instead.